jetbrains-core/tst/software/aws/toolkits/jetbrains/core/execution/AwsConnectionRunConfigurationExtensionSettingsEditorTest.kt

```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    @Test
+    fun canLoadSpecificManualSelection() {
+        val configuration = createConfiguration {
+            useCurrentConnection = false
+            region = "us-east-1"
+            credential = "DUMMY"
+        }
+
+        val editor = AwsConnectionRunConfigurationExtensionSettingsEditor<ApplicationConfiguration>(projectRule.project)
+
+        editor.resetFrom(configuration)
+
+        assertThat(editor.view.manuallyConfiguredConnection.isSelected).isTrue()
+        assertThat(editor.view.region.isEnabled).isTrue()
+        assertThat(editor.view.region.itemCount).isGreaterThan(0)
+        assertThat(editor.view.credentialProvider.isEnabled).isTrue()
+        assertThat(editor.view.credentialProvider.itemCount).isGreaterThan(0)
+
+        assertThat(editor).isRoundTripped {
+            useCurrentConnection = false
+            region = "us-east-1"
+            credential = "DUMMY"
+        }
+    }
+
+    @Test
+    fun canLoadNone() {
+        val configuration = createConfiguration { }
+
+        val editor = AwsConnectionRunConfigurationExtensionSettingsEditor<ApplicationConfiguration>(projectRule.project)
+
+        editor.resetFrom(configuration)
+
+        assertThat(editor.view.none.isSelected).isTrue()
+
+        assertThat(editor.view.credentialProvider.isEnabled).isFalse()
+        assertThat(editor.view.credentialProvider.itemCount).isZero() // We don't want to eagerly load for every RunConfiguration
+
+        assertThat(editor.view.region.isEnabled).isFalse()
+        assertThat(editor.view.region.itemCount).isZero() // We don't want to eagerly load for every RunConfiguration
+
+        assertThat(editor).isRoundTripped {
+            useCurrentConnection = false
+            region = null
+            credential = null
+        }
+    }
+
+    @Test
+    fun manualConnectionEnablesDropDowns() {
+        val editor = AwsConnectionRunConfigurationExtensionSettingsEditor<ApplicationConfiguration>(projectRule.project)
+        editor.view.manuallyConfiguredConnection.doClick()
+
+        assertThat(editor.view.region.isEnabled).isTrue()
+        assertThat(editor.view.credentialProvider.isEnabled).isTrue()
+    }
+
+    private fun createConfiguration(optionBuilder: AwsConnectionRunConfigurationExtensionOptions.() -> Unit): ApplicationConfiguration {
+        val runManager = RunManager.getInstance(projectRule.project)
+        val configuration = runManager.createConfiguration("test", ApplicationConfigurationType::class.java).configuration as ApplicationConfiguration
+        configuration.putCopyableUserData(AWS_CONNECTION_RUN_CONFIGURATION_KEY, AwsConnectionRunConfigurationExtensionOptions().apply(optionBuilder))
+        return configuration
+    }
+
+    private fun ApplicationConfiguration.extensionOptions() = getCopyableUserData(AWS_CONNECTION_RUN_CONFIGURATION_KEY)
+
+    private fun ObjectAssert<AwsConnectionRunConfigurationExtensionSettingsEditor<ApplicationConfiguration>>.isRoundTripped(
```

Comment:
I am trying to understand this round trip

1. It makes an empty config
2. It applies what to the config? It seems its the ObjectAssert? What does that do?
3. It then makes a dummy config from the passed in `expected` block and compares it field by field to the updatedConfiguration

I am not understanding the 2nd line

Reply:
The assertion basically takes the current editor state, applies it to an empty configuration (which effectively copies it into the `putCopyableUserData`) and then asserts that when we pull it back out again we get the same object.

It's a badly named method - I've renamed it to `isPersistedAs`